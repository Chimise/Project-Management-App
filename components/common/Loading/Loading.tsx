import React from "react";
import Spinner, { SpinnerProps } from "../../ui/Spinner";
import cn from "classnames";

interface LoadingProps extends SpinnerProps {
    spinnerClassName?: string;
}

const Loading = ({ className, spinnerClassName, ...props }: LoadingProps) => {
  return (
    <div className={cn("w-full flex items-center justify-center", className)}>
      <Spinner className={spinnerClassName} {...props} />
    </div>
  );
};

export default Loading;
