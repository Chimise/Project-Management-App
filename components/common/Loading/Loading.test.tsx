import "@testing-library/jest-dom";
import {
  render,
} from "../../../test/test-utils";
import router, {
  mockUseRouter,
} from "../../../test/router";
import Loading from './Loading';

beforeEach(() => {
  jest.useFakeTimers();
  jest.clearAllMocks();
});

afterEach(() => {
  jest.useRealTimers();
});

mockUseRouter();



test('It renders unchanged', () => {
    const {container} = render(<Loading />)
    expect(container.firstChild).toMatchSnapshot();
})
