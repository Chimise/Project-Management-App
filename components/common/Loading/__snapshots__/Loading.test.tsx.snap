// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders unchanged 1`] = `
<div
  class="w-full flex items-center justify-center"
>
  <div
    class="relative rounded-full before:absolute before:inset-0 before:rounded-full after:absolute after:inset-0 after:rounded-full after:border-transparent after:animate-spin w-8 h-8 before:border-gray-300 after:border-b-slate-700 before:border-2 after:border-2"
  />
</div>
`;
